vaccine_feed_ingest/runners/il/juvare/normalize.py

```diff
@@ -0,0 +1,256 @@
+    # NOTE: some locations are missing the ZIP code. We return None for these
+    # locations because the schema requires a ZIP code.
+
+    # The street name often runs together with the city name, like this:
+    # "3601 W 183rd St Hazel Crest IL 60429".
+    # We look for punctuation or a street type (like "St") at the end of the
+    # street name.
+
+    match = re.match(
+        r"""
+        (?P<building>[^0-9]*)        # building name
+        (?P<street>[0-9].*           # street address, must end with punctuation or street type
+            (\b(ave|avenue|blvd|boulevard|cir|circle|ct|court|dr|drive|hwy|highway|ln|lane|pkwy|parkway|st|street|way)\b\s*|[.,0-9]\s*|[\r\n])
+        )
+        (?P<city>(\b\w+[ ]*)+)       # city
+        [,]?\s+(IL|Illinois)\b[.,]?  # state
+        \s+(?P<zip>\d{5})(-\d{4})?   # zip
+        (?P<extra>.*)                # building name, instructions, county name, "united states", etc.
+        """,
+        site["location"],
+        re.DOTALL | re.IGNORECASE | re.VERBOSE,
+    )
+
+    if not match:
+        return ("", None)
+
+    address = schema.Address(
+        street1=match.group("street").strip(" ,\r\n"),
+        city=match.group("city"),
+        state="IL",
+        zip=match.group("zip"),
+    )
+    building = match.group("building") or match.group("extra")
+    if re.match(
+        r".*\b(county|united states|enter off crystal lake)\b",
+        building,
+        flags=re.IGNORECASE,
+    ):
+        building = ""
+    building = building.strip(" ,\r\n@.")
+    return (building, address)
+
+
+def _get_contact(site: dict) -> schema.Contact:
+    # indirect_url = "https://covidvaccination.dph.illinois.gov/"
+
+    # NOTE: This direct url bypasses the consent form used at the indirect url,
+    # but it's very easy for people to find the direct url by following links
+    # from provider websites and search results, so it should be okay to return
+    # it.
+    direct_url = f"https://events.juvare.com/{site['organizer']}/{site['slug']}/"
+
+    return [schema.Contact(contact_type="booking", website=direct_url)]
+
+
+def _get_inventory(site: dict) -> Optional[List[schema.Vaccine]]:
+    name = site["name"]
+
+    inventory = []
+
+    pfizer = re.search("pfizer", name, re.IGNORECASE)
+    moderna = re.search("moderna", name, re.IGNORECASE)
+    johnson = re.search("janssen|johnson.*johnson|j&j", name, re.IGNORECASE)
+
+    if pfizer:
+        inventory.append(
+            schema.Vaccine(vaccine="pfizer_biontech", supply_level="in_stock")
+        )
+    if moderna:
+        inventory.append(schema.Vaccine(vaccine="moderna", supply_level="in_stock"))
+    if johnson:
+        inventory.append(
+            schema.Vaccine(vaccine="johnson_johnson_janssen", supply_level="in_stock")
+        )
+
+    if len(inventory) == 0:
+        return None
+
+    return inventory
+
+
+def _filter_name(building: str, site: dict) -> str:
```

Comment:
minor: Can you provide some examples of original address -> name so I know the context of this?

Reply:
Done.